fix: Add parameter validation for removeObject and getObject

nitisht opened the pull request:
fixes #651

balamurugana commented on api/src/main/java/io/minio/MinioClient.java:
```diff
@@ -1505,8 +1505,15 @@ public String getObjectUrl(String bucketName, String objectName)
+    
+    if ((objectName == null) || (objectName.isEmpty())) {
+      throw new InvalidArgumentException("object name cannot be empty");
+    }
```
Add this check at line #1849 and #1581

nitisht replied:
updated @balamurugana

balamurugana replied:
If so, why this check is here?

balamurugana commented on api/src/main/java/io/minio/MinioClient.java:
```diff
@@ -1586,6 +1593,14 @@ public InputStream getObject(String bucketName, String objectName, long offset,
+    
+    if ((objectName == null) || (objectName.isEmpty())) {
+      throw new InvalidArgumentException("object name cannot be empty");
+    }
```
Do this validation first.  See the order of method arguments.

nitisht commented:
I get your point now @balamurugana . Updated the PR to add checks only for final `getObject` and `removeObject` methods.  Removed the checks from wrapper methods

balamurugana commented on api/src/main/java/io/minio/MinioClient.java:
```diff
@@ -1576,16 +1575,23 @@ public InputStream getObject(String bucketName, String objectName, long offset)
       throw new InvalidArgumentException("length should be greater than zero");
     }
-
+      
```
fix this whitespace

nitisht replied:
done
